contracts/SmartSessionCompatibilityFallback.sol

```diff
@@ -53,12 +57,15 @@ contract SmartSessionCompatibilityFallback is ERC7579FallbackBase {
             uint256[] memory extensions
         )
     {
-        //@dev The way that the nested EIP712 logic is supposed to include the address of the account in the
-        // hash is through the verifyingContract field.
+        
+        verifyingContract = msg.sender;
 
-        (fields, name, version, chainId,, salt, extensions) = EIP712(SMART_SESSION_IMPL).eip712Domain();
+        // follows "vendorname.accountname.semver" structure as per ERC-7579
+        string memory accountId = IERC7579Account(msg.sender).accountId();
+        
+        //parse name from accountId
+        (name, version) = accountId.parseAccountId();
 
-        // forcefully overwrite the verifyingContract field with the account address
-        verifyingContract = msg.sender;
+        chainId = block.chainid;
```

Comment:
Note that the `fields` value returned from `eip712Domain()` [is defined as follows](https://eips.ethereum.org/EIPS/eip-5267#specification):

> `fields`: A bit map where bit `i` is set to 1 if and only if domain field `i` is present (`0 ≤ i ≤ 4`). Bits are read from least significant to most significant, and fields are indexed in the order that is specified by EIP-712, identical to the order in which they are listed in the function type.

Since the `SmartSessionCompatibilityFallback` sets the `name`, `version`, `chainId` and `verifyingContract`, but doesn't set the `salt` value, it'd make sense to return `fields` as `0b01111` which equals `hex"0f"`:
```diff
function eip712Domain()
    public
    view
    returns (
        bytes1 fields,
        string memory name,
        string memory version,
        uint256 chainId,
        address verifyingContract,
        bytes32 salt,
        uint256[] memory extensions
    )
{
+   fields = hex"0f"; // 01111    
    verifyingContract = msg.sender;

    // follows "vendorname.accountname.semver" structure as per ERC-7579
    string memory accountId = IERC7579Account(msg.sender).accountId();
    
    //parse name from accountId
    (name, version) = accountId.parseAccountId();

    chainId = block.chainid;
}
```

Also note that this matches the behavior from the [OpenZeppelin `EIP712.sol` code](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7bd2b2aaf68c21277097166a9a51eb72ae239b34/contracts/utils/cryptography/EIP712.sol#L129), which was being used in `SmartSessionERC7739` before

Reply:
Makes sense! thank you very much for this catch